core/src/main/java/hudson/security/csrf/DefaultCrumbIssuer.java

```diff
@@ -57,7 +57,7 @@ private Object readResolve() {
         return this;
     }
 
-    private void initializeMessageDigest() {
+    private synchronized void initializeMessageDigest() {
```

Comment:
Probably only used by XStream, so synchronization is technically unnecessary, but it seems only logical to synchronize given that this method is setting `md` and readers of `md` should not be able to see it in a partially initialized state.